### PR TITLE
STITCH-1342 Add PromiseKit extensions for v4 iOS SDK

### PR DESCRIPTION
**Pull request by jsflax**

### CODE REVIEW

**Comment by jsflax** [comment]
Note, I have to change some of the docstrings. Getting on that now.

File: StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/StitchAppClient+PMK.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+     *     - withName: The name of the Stitch function to be called.
+     *     - withArgs: The `BSONArray` of arguments to be provided to the function.
+     *     - completionHandler: The completion handler to call when the function call is complete.
+     *                          This handler is executed on a non-main global `DispatchQueue`.
+     *     - result: The result of the function call as an `Any`, or `nil` if the function call failed.
+     *     - error: An error object that indicates why the function call failed, or `nil` if the function call was
+     *              successful.
+     *
+     */
+    func callFunction(withName name: String, withArgs args: BSONArray) -> Promise<Any> {
```
Comment (adamchel) [inline StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/StitchAppClient+PMK.swift]: Not really an issue with your PR, but this is going to change soon when we do Codable stuff. 

In the interest of this not getting out of sync with the iOS module, should we agree now that updating the PromiseKit extensions when we add or remove features should be just be one of the things we always do with any PR like tests and comments?

File: StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/Adapters.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import PromiseKit
+
+// swiftlint:disable identifier_name
```
Comment (adamchel) [inline StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/Adapters.swift]: [nit] we could just rename `t` and `e` to something like `value` and `error` to be more readable in general

File: StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/Stitch_PromiseKit.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (adamchel) [inline StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/Stitch_PromiseKit.h]: [nit] Remove

**Comment by adamchel** [comment]
My LGTM is also pending updated docstrings

File: StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/StitchAppClient+PMK.swift [hunk 4]
```diff
@@ -12,9 +12,7 @@ extension StitchAppClient {
      *     - withArgs: The `BSONArray` of arguments to be provided to the function.
      *     - completionHandler: The completion handler to call when the function call is complete.
```
Comment (adamchel) [inline StitchCore-iOS/Extensions/PromiseKit/Stitch-PromiseKit/Stitch-PromiseKit/StitchAppClient+PMK.swift]: [nit] remove

**Pull request merged**